src/ImageFitting/ImageFitter.cc

```diff
@@ -268,9 +307,11 @@ void ImageFitter::CalculateErrors() {
     }
 
     for (size_t i = 0; i < _num_components; i++) {
-        double center_x, center_y, amp, fwhm_x, fwhm_y, pa;
-        std::tie(center_x, center_y, amp, fwhm_x, fwhm_y, pa) =
-            GetGaussianParams(_fit_values, i * 6, _fit_data.fit_values_indexes, _fit_data.initial_values, 0, 0);
+        GaussianParams params = GetGaussianParams(_fit_values, i * 6, _fit_data.fit_values_indexes, _fit_data.initial_values, 0, 0);
+        double amp = params.amp;
+        double fwhm_x = params.fwhm_x;
+        double fwhm_y = params.fwhm_y;
+        double pa = params.pa;
```

Comment:
You can use a structured binding for this unpacking instead:
```
auto [center_x, center_y, amp, fwhm_x, fwhm_y, pa] = GetGaussianParams(...);
```

(But see also the previous comment about turning that function into a constructor.)